cmd/common.go

```diff
@@ -9,47 +9,31 @@ import (
 	"github.com/odpf/meteor/metrics"
 	"github.com/odpf/meteor/plugins"
 	plugin "github.com/odpf/meteor/plugins/external"
-	extractorPlugins "github.com/odpf/meteor/plugins/extractors"
-	processorPlugins "github.com/odpf/meteor/plugins/processors"
-	sinkPlugins "github.com/odpf/meteor/plugins/sinks"
 )
 
 func initRunner(config config.Config, logger plugins.Logger) (runner *recipe.Runner, cleanFn func()) {
-	extractorFactory := initExtractorFactory(logger)
 	processorFactory, killPluginsFn := initProcessorFactory()
-	sinkFactory := initSinkFactory()
 	metricsMonitor := initMetricsMonitor(config)
 	runner = recipe.NewRunner(
-		extractor.New(extractorFactory),
+		extractor.Catalog,
 		processorFactory,
-		sinkFactory,
+		sink.Catalog,
 		metricsMonitor,
 	)
 	cleanFn = func() {
 		killPluginsFn()
 	}
 	return
 }
-func initExtractorFactory(logger plugins.Logger) *extractor.Factory {
-	factory := extractor.NewFactory()
-	extractorPlugins.PopulateFactory(factory, logger)
-	return factory
-}
 func initProcessorFactory() (*processor.Factory, func()) {
```

Comment:
Can this be encapsulated within processor.Catalog to make programming interfaces consistent with sink and extractor.

Reply:
oh yeah, this is a casualty of merge conflict 😭

Reply:
Nevermind, this was done on purpose actually. I left it unchanged as I need to refactor how runtime plugins are being instantiated. Because runtime plugins are currently broken so didn't fix it. I'll do it later once rest of the implementation looks concrete. For now, I'll just disable runtime plugins completely?

Reply:
@StewartJingga ^^

Reply:
I have disabled runtime plugins in this PR for now.